src/library/export/dlglibraryexport.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <memory>
+
+#include <QDialog>
+#include <QLineEdit>
+#include <QListWidget>
+#include <QRadioButton>
+#include <QTreeWidget>
+#include <QWidget>
+
+#include "library/crate/crateid.h"
+#include "library/export/engineprimeexportrequest.h"
+#include "preferences/usersettings.h"
+#include "util/parented_ptr.h"
+
+class TrackCollectionManager;
+
+namespace mixxx {
+
+/// The DlgLibraryExport class is a UI window that gathers information from
+/// the user about how they would like to export the Mixxx library.
+///
+/// Currently, the dialog only supports exporting to the Engine Library format,
+/// but in future it is expected that this dialog could be expanded to include
+/// other formats, and generate different export signals accordingly.
+class DlgLibraryExport : public QDialog {
+    Q_OBJECT
+
+  public:
+    DlgLibraryExport(
+            QWidget* parent, UserSettingsPointer pConfig, TrackCollectionManager& trackCollectionManager);
+    void setSelectedCrate(CrateId crateId);
+
+  signals:
+    /// The startEnginePrimeExport signal is emitted when sufficient information
+    /// has been gathered from the user to kick off an Engine Prime export, and
+    /// details of the request are provided as part of the signal.
+    void startEnginePrimeExport(EnginePrimeExportRequest) const;
+
+  private slots:
+    void exportWholeLibrarySelected();
+    void exportSelectedCratedSelected();
+    void browseExportDirectory();
+    void exportRequested();
+
+  private:
+    UserSettingsPointer m_pConfig;
+    TrackCollectionManager& m_trackCollectionManager;
```

Comment:
Store a pointer instead of a mutable reference

Reply:
What's the reasoning for this?  A pointer obviously opens up the possibility that it might be `nullptr`, no?

Reply:
Happy to update as per the style guide.